lib/browser/methods/ie.rb

```diff
@@ -2,6 +2,38 @@ class Browser
   module IE
     TRIDENT_VERSION_REGEX = %r[Trident/([0-9.]+)]
     MODERN_IE = %r[Trident/.*?; rv:(.*?)]
+    MSIE = %r{MSIE ([\d.]+)|Trident/.*?; rv:([\d.]+)}
+    EDGE = %r{(Edge/[\d.]+|Trident/8)}
+
+    # https://msdn.microsoft.com/en-us/library/ms537503(v=vs.85).aspx#TriToken
+    TRIDENT_MAPPING = {
+      "4.0" => "8",
+      "5.0" => "9",
+      "6.0" => "10",
+      "7.0" => "11",
+      "8.0" => "12"
+    }
+
+    def ie_version
+      TRIDENT_MAPPING[trident_version] || msie_version
+    end
+
+    def ie_full_version
+      "#{ie_version}.0"
+    end
+
+    def msie_full_version
+      ua.match(MSIE) && ($1 || $2)
+    end
+
+    def msie_version
+      msie_full_version.split('.').first
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.